File: Python/ShortestWord.py
# Level: 7kyu
# Name: Shortest Word
# Instructions: Simple, given a string of words, return the length of the shortest word(s).
# String will never be empty and you do not need to account for different data types.

# Solution 1 in Python


def find_short(s):
    return min(len(x) for x in s.split())

# Solution 2 in Python


def find_short(s):
    return len(sorted(s.split(' '), key=len)[0])

# Sample Tests Passed:
# test.describe("Basic Tests")
# test.assert_equals(find_short("bitcoin take over the world maybe who knows perhaps"), 3)
# test.assert_equals(find_short("turns out random test cases are easier than writing out basic ones"), 3)
# test.assert_equals(find_short("lets talk about javascript the best language"), 3)
# test.assert_equals(find_short("i want to travel the world writing code one day"), 1)
# test.assert_equals(find_short("Lets all go on holiday somewhere very cold"), 2)

File: Javascript/ShortestWord.js
// Language: Javascript
// Level: 7kyu
// Name of Problem: Shortest Word

// Instructions: Simple, given a string of words, return the length of the shortest word(s).
// String will never be empty and you do not need to account for different data types.

// Solution 1 using split(), sort(), and pop():
function findShort(s){
  return s.split(' ').sort((a,b) => (b.length - a.length)).pop().length
}

// Solution 2:
function findShort(s){
  let arr = s.split(' ')
  let shortest = arr[0]
    for (let i = 0; i < arr.length; i++) {
    if (arr[i].length < shortest.length) {
  shortest = arr[i]
    }
  }
return shortest.length;
}

// Solution 3 using Infinity:
const findShort = s => {
  let wordArr = s.split(' ')
  let shortest = Infinity
  for (let i = 0; i < wordArr.length; i++) {
    let wordLength = wordArr[i].length
    if (wordLength < shortest) {
      shortest = wordLength
    }
  }
  return shortest
}

// Sample Tests Passed:
// Test.describe("Example tests",_=>{
//   Test.assertEquals(findShort("bitcoin take over the world maybe who knows perhaps"), 3);
//   Test.assertEquals(findShort("turns out random test cases are easier than writing out basic ones"), 3);
//   });